velox/functions/prestosql/aggregates/ReservoirSampleAggregate.cpp

```diff
@@ -0,0 +1,663 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/prestosql/aggregates/ReservoirSampleAggregate.h"
+#include "velox/functions/prestosql/aggregates/ReservoirSampleAccumulator.h"
+#include "velox/type/Type.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::aggregate::prestosql {
+
+namespace {
+
+class ReservoirSampleAggregate : public exec::Aggregate {
```

Comment:
nit: For such long classes, it helps if the class is defined first with its methods and then the implementations follow outside the class definition .